handling-forms/controlled-forms.md

```diff
@@ -415,9 +422,71 @@ Each new student should have a name and an email. Her form contains one text inp
 
 ### !callout-info
 
-## Labels and Names
+## Labels and Inputs and Their Attributes
+
+What's up with the attributes `name`, `id` , and `for` (mapped in React as `htmlFor`)? These attributes are used by HTML forms when submitting data and for improving a web page's accessibility. 
+
+<br/>
+
+The `id` and `for` attributes enable us to [associate a `<label>` element with its corresponding `<input>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#labels_2) which improves accessibility because assistive technologies can tell what the input is for and the area a user can click is increased from just the input element to include the label element too. 
+
+<br/>
+
+The [name attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) on an `<input>` element assigns a name to tit and this name can be used during form submission or as a reference for other purposes.
+
+<br/>
+
+We know that we need to include the `id` attribute for accessibility reasons, but what if we want to use the same component on a page two times? If the same component appears in an application more than once then the `id`, which is meant to be a unique identifier, would also show up more than once. In this scenario, the `id` would not be unique in the entire DOM.
+
+<br/>
+
+Expand below to learn how we can ensure `id`s are unique in a React app.
+
+<br/>
+
+<details>
```

Comment:
Wasn't sure if this infor should be in a <detail> or not but the callout felt like it was getting long. What do you think?